src/iam.ts

```diff
@@ -262,14 +266,37 @@ export class IAM extends IAMBase {
   async updateDidDocument(options: {
     didAttribute: DIDAttribute;
     data: IUpdateData;
+    did?: string;
     validity?: number;
   }): Promise<boolean> {
-    const { didAttribute, data, validity } = options;
-    if (this._document) {
+    const { didAttribute, data, validity, did } = options;
+
+    if (!this._document) {
+      throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+    }
+
+    if (!this._didSigner) {
+      throw new Error(ERROR_MESSAGES.SIGNER_NOT_INITIALIZED);
+    }
+
+    if (!did) {
       const updated = await this._document.update(didAttribute, data, validity);
       return Boolean(updated);
     }
-    throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
+
+    const defaultValue: IAttributePayload = { publicKey: `0x${new Keys().publicKey}` };
+
+    const updateData: IUpdateData = {
+      algo: Algorithms.Secp256k1,
+      encoding: Encoding.HEX,
+      ...data,
+      value: { ...defaultValue, ...data.value }
+    };
```

Comment:
Why not just using the `IUpdateData` that was passed in as a param?

Reply:
Looking on task description, some of the values should be predefined. For example 

> the key should be of type “sigAuth” 

But yeah, i can remove it.

Reply:
Ah yea, you're right, my fault. I think though that we can push the setting of those values up to Switchboard Dapp interaction with iam-client-lib (i.e. specify these values in https://energyweb.atlassian.net/browse/SWTCH-895) and keep this function generic.

Reply:
The problem with not predefined data is that, we have optional parameters in `IUpdateData`. If we do not specify them (those optionals), then we can get an error. Function is still generic, because we use predefined data, but you can override them - i'm merging two objects.